src/platform/cert_capi.c

```diff
@@ -873,21 +874,27 @@ CxPlatCertValidateChain(
 
     CERT_CHAIN_PARA ChainPara;
 
-    static const LPSTR UsageOids[] = {
+    static const LPSTR ServerUsageOids[] = {
         szOID_PKIX_KP_SERVER_AUTH,
         szOID_SERVER_GATED_CRYPTO,
         szOID_SGC_NETSCAPE
     };
 
+    static const LPSTR ClientUsageOids[] =  {
+        szOID_PKIX_KP_CLIENT_AUTH
+    };
+
     if (ValidationError != NULL) {
         *ValidationError = NO_ERROR;
     }
 
     memset(&ChainPara, 0, sizeof(ChainPara));
     ChainPara.cbSize = sizeof(ChainPara);
     ChainPara.RequestedUsage.dwType = USAGE_MATCH_TYPE_OR;
-    ChainPara.RequestedUsage.Usage.cUsageIdentifier = ARRAYSIZE(UsageOids);
-    ChainPara.RequestedUsage.Usage.rgpszUsageIdentifier = (LPSTR*)UsageOids;
+    ChainPara.RequestedUsage.Usage.cUsageIdentifier =
+        (CredFlags & QUIC_CREDENTIAL_FLAG_CLIENT) ? ARRAYSIZE(ClientUsageOids) : ARRAYSIZE(ServerUsageOids);
+    ChainPara.RequestedUsage.Usage.rgpszUsageIdentifier =
+        (CredFlags & QUIC_CREDENTIAL_FLAG_CLIENT) ? (LPSTR*)ClientUsageOids : (LPSTR*)ServerUsageOids;
```

Comment:
Are these swapped too? Clients should validate the certs are used for server auth, right?